docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,350 @@
+[[migrate_to_java_time]]
+=== Java time migration guide
+
+With 7.0, {es} switch to java time from joda time for date-related parsing,
+formatting, and calculations. This guide is designed to help you determine
+if your cluster is impacted and, if so, prepare for the upgrade.
+
+Joda style indices are supported in 6.8, but indices created with joda-style patterns in 6.x are also allowed to be used in 7.0
+All new indices created in 7.x are expected to be using java-time patterns.
+
+We advice to prepare for this migration in 6.8, but it will also work if you decide to perform it in version 7.x
+
+- In 6.8 you need to prefix your new java.style pattern with `8`. Patterns without `8` are considered joda style.
+- In 7.x patterns without `8` on indices *created in 6.x* are considered joda-style.
+- Patterns with or without `8` on indices *created in 7.x*  are considered java.style
+
+==== Impacted features
+The switch to java time only impacts custom <<date,`date`>> and
+<<date_nanos,`date_nanos`>> formats.
+
+These formats are commonly used in:
+
+* <<mapping,Index mappings>>
+* <<indices-templates,Index templates>>
+* <<pipeline,Ingest pipelines>>
+
+If you don't use custom date formats, you can skip the rest of this guide.
+Most of the custom date formats are compatible, but with some there is a change of meaning or missing. More on that below.
+
+To see if your date format is impacted, use the <<migration-api-deprecation,deprecation info API>>
+or the {kibana-ref}/upgrade-assistant.html[Kibana upgrade assistant]
+
+
+==== Incompatible formats
+Custom date formats containing the following joda time literals should be
+changed before upgrading.
+
+`Y` (Year of era)::
++
+--
+Replace with `y`.
+
+*Example:*
+`YYYY-MM-dd` should become `yyyy-MM-dd`
+
+In java time, `Y` is used for
+https://docs.oracle.com/javase/8/docs/api/java/time/temporal/WeekFields.html[week-based year].
+Using `Y` in place of `y` could result in off-by-one errors in year calculation.
+
+
+The difference between year-of-era and week based year
+For pattern `YYYY-ww` and date `2019-01-01T00:00:00.000Z`  will give `2019-01`
+For pattern `YYYY-ww` and date `2018-12-31T00:00:00.000Z`  will give `2019-01` (counter intuitive) because there is >4 days of that week in 2019
+--
+
+`y` year::
++
+--
+Replace with `u`.
+
+*Example:*
+`yyyy-MM-dd` should become `uuuu-MM-dd`
+
+The difference is that year can contain non positive values, whereas `y` can not. Also year-of-era can also be associated with an era field.
+--
+
+
+`C` century of era::
++
+--
+No longer supported in 7.0. There is no replacement, so you need to preprocess your input.
+
+A possible workaround https://stackoverflow.com/questions/38354151/how-to-force-java-time-localdate-to-assume-19th-century-as-yy-year-pattern
```

Comment:
I don't think we need to include the stackoverflow link, but I'll defer to you if you feel strongly.

I think a tip on ways to preprocess your input (for example, using an ingest processor) would be great.
```suggestion
Century of era is not supported in java time or {es} 7.0.
There is no replacement. Instead, we recommend you preprocess your input.
```

Reply:
agree, it is not a 100% workaround so better to suggest a preprocessing. 
The scenario is very uncommon. I guess an example processor or pipeline would unnecessarily extend this guide. I will stick with your suggestion.